modules/core/src/main/java/org/apache/ignite/internal/processors/affinity/GridAffinityAssignmentCache.java

```diff
@@ -304,6 +309,60 @@ public void onReconnected() {
     }
 
     /**
+     * @param assignment List indexed by partition number.
+     * @param nodesCnt Nodes count.
+     */
+    private void printDistribution(List<List<ClusterNode>> assignment, int nodesCnt) {
+        Float ignitePartDistribution = getFloat(IGNITE_PART_DISTRIBUTION_WARN_THRESHOLD, 0.1f);
+
+        int[] partitionsByLocalNode = new int[nodesCnt];
+
+        int totalPrimaryCnt = 0;
+
+        int totalBackupCnt = 0;
+
+        for (int i = 0; i < nodesCnt; i++) {
```

Comment:
Whole loop looks strange.
We have List of List, but task nodesCnt as method parameter and using it while iteration throw List of List.
Can you describe algorithm?